src/Microsoft.DotNet.XHarness.iOS.Shared/Hardware/SimulatorDevice.cs

```diff
@@ -201,4 +201,28 @@ public async Task<bool> Boot(ILog log, CancellationToken cancellationToken)
 
         return true;
     }
+
+    public async Task<string> GetAppBundlePath(ILog log, string bundleIdentifier, CancellationToken cancellationToken)
+    {
+        log.WriteLine($"Querying '{Name}' for bundle path of '{bundleIdentifier}'..");
+
+        var output = new MemoryLog() { Timestamp = false };
+        var result = await _processManager.ExecuteXcodeCommandAsync(
+            "simctl",
+            new[] { "get_app_container", UDID, bundleIdentifier },
+            log,
+            output,
+            output,
+            TimeSpan.FromSeconds(30));
+
+        if (!result.Succeeded)
+        {
+            throw new Exception($"Failed to get information for '{bundleIdentifier}'. Please check the app is installed");
```

Comment:
Maybe we should add that the app failed to get the information in 30 seconds, in case it was a time out?

Reply:
The logs from the `_processManager` call above will actually have that info if people want to dig into the real reason (from a high-level  end user standpoint we don't care much)